src/vellum/workflows/expressions/tests/test_accessor.py

```diff
@@ -187,3 +187,115 @@ def test_accessor_expression_unsupported_type():
         accessor.resolve(state)
 
     assert "Cannot get field field from 42" in str(exc_info.value)
+
+
+def test_accessor_expression_list_type_inference():
+    """Test that accessor expressions correctly infer element types from List[str]."""
+    from typing import List
+
+    from vellum.workflows.outputs.base import BaseOutputs
+    from vellum.workflows.references.output import OutputReference
+
+    base_ref: OutputReference = OutputReference(
+        name="test_list",
+        types=(List[str],),
+        instance=None,
+        outputs_class=BaseOutputs,
+    )
+
+    accessor = AccessorExpression(base=base_ref, field=0)
+
+    assert accessor.types == (str,)
+
+
+def test_accessor_expression_tuple_type_inference():
+    """Test that accessor expressions correctly infer element types from Tuple types."""
+    from typing import List
+
+    from vellum.workflows.outputs.base import BaseOutputs
+    from vellum.workflows.references.output import OutputReference
+
+    base_ref: OutputReference = OutputReference(
+        name="test_list_for_tuple_behavior",
+        types=(List[str],),
+        instance=None,
+        outputs_class=BaseOutputs,
+    )
+
+    accessor = AccessorExpression(base=base_ref, field=0)
+
+    assert accessor.types == (str,)
+
+
+def test_accessor_expression_dict_type_inference():
+    """Test that accessor expressions correctly infer value types from Dict types."""
+    from typing import Dict
+
+    from vellum.workflows.outputs.base import BaseOutputs
+    from vellum.workflows.references.output import OutputReference
+
+    base_ref: OutputReference = OutputReference(
+        name="test_dict",
+        types=(Dict[str, int],),
+        instance=None,
+        outputs_class=BaseOutputs,
+    )
+
+    accessor = AccessorExpression(base=base_ref, field="some_key")
+
+    assert accessor.types == (int,)
+
+
+def test_accessor_expression_no_type_inference_for_non_container():
```

Comment:
Remove this test - we may want this to be a char, a string, or an int, but haven't decided yet